ephios/plugins/federation/forms.py

```diff
@@ -0,0 +1,102 @@
+import base64
+import binascii
+import json
+from json import JSONDecodeError
+from urllib.parse import urljoin
+
+import requests
+from django import forms
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.forms import CheckboxSelectMultiple
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from dynamic_preferences.registries import global_preferences_registry
+from requests import HTTPError, ReadTimeout
+
+from ephios.api.models import Application
+from ephios.core.forms.events import BasePluginFormMixin
+from ephios.plugins.federation.models import (
+    FederatedEventShare,
+    FederatedGuest,
+    FederatedHost,
+    InviteCode,
+)
+
+
+class EventAllowFederationForm(BasePluginFormMixin, forms.Form):
+    shared_with = forms.ModelMultipleChoiceField(
+        queryset=FederatedGuest.objects.all(), required=False, widget=CheckboxSelectMultiple
+    )
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("prefix", "federation")
+        self.event = kwargs.pop("event")
+        self.request = kwargs.pop("request")
+        super().__init__(*args, **kwargs)
+        try:
+            self.instance = FederatedEventShare.objects.get(event_id=self.event.id)
+        except (AttributeError, FederatedEventShare.DoesNotExist):
+            self.instance = FederatedEventShare(event=self.event)
+        self.fields["shared_with"].initial = (
+            self.instance.shared_with.all() if self.instance.pk else []
+        )
+
+    def save(self):
+        if self.cleaned_data["shared_with"] and not self.instance.pk:
+            self.instance.save()
+        if self.instance.pk:
+            self.instance.shared_with.set(self.cleaned_data["shared_with"])
+
+    @property
+    def heading(self):
+        return _("Share event with other ephios instances")
+
+    def is_function_active(self):
+        return self.instance.pk and self.instance.shared_with.exists()
+
+
+class InviteCodeForm(forms.ModelForm):
+    class Meta:
+        model = InviteCode
+        fields = ["url"]
```

Comment:
Make sure all the labels are translated, either in the model fields for model forms or right here as label on form fields. It should be "URL", not "Url" as a label.

Reply:
Also I think some placeholder would be great to indicate that you must input `_("https://their-name.ephios.de")`